lambdas/check-gpg45-score/src/main/java/uk/gov/di/ipv/core/checkgpg45score/CheckGpg45ScoreHandler.java

```diff
@@ -0,0 +1,160 @@
+        this.configService = new ConfigService();
+        this.clientOAuthSessionDetailsService = new ClientOAuthSessionDetailsService(configService);
+        this.gpg45ProfileEvaluator = new Gpg45ProfileEvaluator(configService);
+        this.ipvSessionService = new IpvSessionService(configService);
+        this.userIdentityService = new UserIdentityService(configService);
+
+        componentId = configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public Map<String, Object> handleRequest(ProcessRequest event, Context context) {
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(event);
+            String featureSet = RequestHelper.getFeatureSet(event);
+            String scoreType = RequestHelper.getScoreType(event);
+            Integer scoreThreshold = RequestHelper.getScoreThreshold(event);
+            configService.setFeatureSet(featureSet);
+
+            int scoreToCompare = getScore(ipvSessionId, scoreType);
+            if (scoreToCompare >= scoreThreshold) {
+                LOGGER.info(
+                        new StringMapMessage()
+                                .with(LOG_MESSAGE_DESCRIPTION.getFieldName(), "Score threshold met")
+                                .with(LOG_SCORE_TYPE.getFieldName(), scoreType));
+                return new JourneyResponse(JOURNEY_MET_PATH).toObjectMap();
+            } else {
+                LOGGER.info(
+                        new StringMapMessage()
+                                .with(
+                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                        "Score threshold not met")
+                                .with(LOG_SCORE_TYPE.getFieldName(), scoreType));
+                return new JourneyResponse(JOURNEY_UNMET_PATH).toObjectMap();
+            }
+        } catch (HttpResponseExceptionWithErrorBody e) {
+            LOGGER.error("Received HTTP response exception", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH, e.getResponseCode(), e.getErrorResponse())
+                    .toObjectMap();
+        } catch (ParseException e) {
+            LOGGER.error("Unable to parse GPG45 scores from existing credentials", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.FAILED_TO_PARSE_ISSUED_CREDENTIALS)
+                    .toObjectMap();
+        } catch (UnknownEvidenceTypeException e) {
+            LOGGER.error("Unable to determine type of credential", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.FAILED_TO_DETERMINE_CREDENTIAL_TYPE)
```

Comment:
`FAILED_TO_DETERMINE_CREDENTIAL_TYPE` doesn't seem like quite the right thing, maybe worth adding a new response?

Reply:
So this is thrown by `gpg45ProfileEvaluator.buildScore` which is used across several lambdas and handled in this manner with this error message and code - I'm not sure we want to deviate from that?

Reply:
Fair, happy to leave it.